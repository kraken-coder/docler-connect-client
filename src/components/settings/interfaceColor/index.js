import InterfaceColor from './interfaceColor'

export default InterfaceColor
