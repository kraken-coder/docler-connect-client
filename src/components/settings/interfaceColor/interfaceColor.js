/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { writeItem, setSettings } from '../../../utils/localStorage'
import { loadSettings } from '../../../state/actions/settingsActions'

import '../style.scss'

// eslint-disable-next-line no-shadow
const InterfaceColor = ({ loadSettings }) => {
  const [state, setstate] = useState('')
  // set default to the value equals to local storage
  useEffect(() => {
    setSettings('theme', 'theme')
    loadSettings()
  }, [state])

  //  set the choosen theme to local storage
  const handleChange = e => {
    const value = e.target.id
    writeItem('theme', value)
    setstate(value)
  }
  return (
    <div className="settings-items">
      <p className="settings-heading">Interface Color</p>
      <div className="inputGroup">
        <input onChange={handleChange} id="light" name="theme" type="radio" />
        <label htmlFor="light">Light</label>
      </div>
      <div className="inputGroup">
        <input onChange={handleChange} id="dark" name="theme" type="radio" />
        <label htmlFor="dark">Dark</label>
      </div>
    </div>
  )
}

InterfaceColor.propTypes = {
  loadSettings: PropTypes.func.isRequired,
}
const mapDispatchToProps = {
  loadSettings,
}
export default connect(null, mapDispatchToProps)(InterfaceColor)
