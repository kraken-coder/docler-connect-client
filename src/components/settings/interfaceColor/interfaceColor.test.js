import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import interfaceColor from './interfaceColor'

test('Renders Without Crashing', () => {
  const Wrapper = shallow(<interfaceColor />)
  expect(toJSON(Wrapper)).toMatchSnapshot()
})
