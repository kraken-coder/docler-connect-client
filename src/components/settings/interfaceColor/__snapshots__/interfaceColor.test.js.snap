// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Without Crashing 1`] = `<interfaceColor />`;
